libslide/src/bignum/utils.rs

```diff
@@ -10,6 +15,56 @@ pub fn recast_vec<T, U>(v: Vec<T>) -> Vec<U> {
     unsafe { Vec::from_raw_parts(p as *mut U, len, cap) }
 }
 
+// note we cannot use the above recast since we have a user defined type with
+// an undefined way of storing its memory
+// prereqs: tryfrom(or from) is implemented for type coercion and default and clone are implemented
+pub fn recast_user_vec<
+    T: std::clone::Clone,
+    U: std::default::Default + std::clone::Clone + std::convert::TryFrom<T>,
+>(
+    v: Vec<T>,
+) -> Result<Vec<U>, &'static str> {
```

Comment:
nit: more idiomatic to use a where clause
```suggestion
pub fn recast_user_vec<T, U>(v: Vec<T>) -> Result<Vec<U>, &'static str>
where
    T: std::clone::Clone,
    U: std::default::Default + std::clone::Clone + std::convert::TryFrom<T>,
{
```